grakn-graql/src/main/java/ai/grakn/graql/internal/pattern/property/DirectIsaProperty.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.graql.internal.pattern.property;
+
+import ai.grakn.GraknTx;
+import ai.grakn.concept.SchemaConcept;
+import ai.grakn.concept.Thing;
+import ai.grakn.concept.Type;
+import ai.grakn.exception.GraqlQueryException;
+import ai.grakn.graql.Var;
+import ai.grakn.graql.admin.Atomic;
+import ai.grakn.graql.admin.ReasonerQuery;
+import ai.grakn.graql.admin.UniqueVarProperty;
+import ai.grakn.graql.admin.VarPatternAdmin;
+import ai.grakn.graql.internal.gremlin.EquivalentFragmentSet;
+import ai.grakn.graql.internal.gremlin.sets.EquivalentFragmentSets;
+import ai.grakn.graql.internal.reasoner.atom.binary.type.IsaAtom;
+import ai.grakn.graql.internal.reasoner.atom.predicate.IdPredicate;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static ai.grakn.graql.internal.reasoner.utils.ReasonerUtils.getIdPredicate;
+
+/**
+ * Represents the {@code direct-isa} property on a {@link Thing}.
+ * <p>
+ * This property can be queried and inserted.
+ * </p>
+ * <p>
+ * THe property is defined as a relationship between an {@link Thing} and a {@link Type}.
+ * </p>
+ * <p>
+ * When matching, any subtyping is ignored. For example, if we have {@code $bob isa man}, {@code man sub person},
+ * {@code person sub entity} then it only follows {@code $bob isa person}, not {@code bob isa entity}.
+ * </p>
+ *
+ * @author Felix Chapman
+ * @author Jason Liu
+ */
+@AutoValue
+public abstract class DirectIsaProperty extends AbstractVarProperty implements UniqueVarProperty, NamedProperty {
```

Comment:
Would it make sense to make `DirectIsaProperty` and `IsaProperty` have a common parent? They mainly differ by the `type()`/`directType()` function which is pretty much a naming difference. Only the `match` function is different to exclude the subs. Other member functions only differ by type()<->directType() change.